proto/src/code_execution.proto

```diff
@@ -0,0 +1,42 @@
+syntax = "proto3";
+
+package code_execution;
+import "google/protobuf/timestamp.proto";
+
+service TestExecution {
+    rpc QueueTest  (TestExecutionRequest) returns (TestExecutionResponse) {}
+}
+
+/*  VersionedTest represents a unique test.
+    params:
+    - test_id: ID of the test
+    - test_version: Integer representing the version of the test
+*/
+message VersionedTest {
+    int32 test_id = 1;
+    int32 test_version = 2;
+}
+
+/*  TestExecutionRequest represents a request to execute a versioned test.
+    params:
+    - session_id: ID representing the trace session
+    - versioned_test: Message representing a unique test
+    - test_input: String array representing stringified inputs to the test
+    - request_timestamp: Timestramp representing when this test execution was requested by the client
+*/
+message TestExecutionRequest {
+    int32 session_id = 1;
+    VersionedTest versioned_test = 2;
+    repeated string test_input = 3;
+    google.protobuf.Timestamp request_timestamp = 4; 
+}
+
+/*  TestExecutionResponse represents the response sent to client after receiving a TestExecutionRequest.
+    params:
+    - versioned_test: Message representing the unique test that was requested
+    - test_input: String array representing stringified inputs to the requested test
+*/
+message TestExecutionResponse {
+    VersionedTest versioned_test = 1;
+    repeated string test_input = 2;
```

Comment:
Yea that should be fine. I was concerned we wouldn't be able to associate successes with the response when sending in batches/high frequency but since we will probably fire them individually for now should be good. gRPC already has some error handling so we can probably keep this an empty message for now as well and add as we go forward.

Reply:
Changed to an empty message